freetypefont.mod/freetypefont.bmx

```diff
@@ -166,7 +166,41 @@ Type TFreeTypeFont Extends BRL.Font.TFont
 				
 		Local ft_face:Byte Ptr
 
-		If src.Find( "::" )>0
+		If TStream(src)
+			Local stream:TStream = TStream(src)
+			Local data:Byte[1024]
+			Local dataSize:Int
+
+			'backup old stream position to set it back to it afterwards
+			Local oldStreamPos:int = stream.Pos()
+			stream.Seek(0)
+			
+			While Not stream.Eof()
+				If dataSize = data.length
+					data = data[.. dataSize*3/2]
+				EndIf
+				dataSize :+ stream.Read( (Byte Ptr data) + dataSize, data.length - dataSize )
+			Wend
+			If dataSize <> data.length
+				data = data[..dataSize]
+			EndIf
+
+			'restore stream position
+			stream.Seek(oldStreamPos)
+
+			'load data into freetype buffer
+			buf_size = data.length
+			If Not buf_size
+				Return Null
+			EndIf
+			buf = MemAlloc( Size_T(buf_size) )
+			MemCopy( buf, data, Size_T(buf_size) )
+
+			If FT_New_Memory_Face( ft_lib, buf, buf_size, 0, Varptr ft_face )
+				MemFree buf
+				Return Null
+			EndIf
+		ElseIf String(src).Find( "::" )>0

```

Comment:
If `src` isn't a String, this may end badly... :-)

Reply:
It only should if the memory address of a "stringified" object contains "::".